airbyte-integrations/connectors/source-woocommerce/source_woocommerce/source.py

```diff
@@ -0,0 +1,199 @@
+        if not next_page_token:
+            params["orderby"] = self.order_field
+            params["order"] = "asc"
+            if stream_state:
+                params["after"] = stream_state.get(self.cursor_field)
+        return params
+
+    # Parse the stream_slice with respect to stream_state for Incremental refresh
+    # cases where we slice the stream, the endpoints for those classes don't accept any other filtering,
+    # but they provide us with the updated_at field in most cases, so we used that as incremental filtering during the order slicing.
+    def filter_records_newer_than_state(self, stream_state: Mapping[str, Any] = None, records_slice: Mapping[str, Any] = None) -> Iterable:
+        # Getting records >= state
+        if stream_state:
+            for record in records_slice:
+                if record[self.cursor_field] >= stream_state.get(self.cursor_field):
+                    yield record
+        else:
+            yield from records_slice
+
+class NonFilteredStream(IncrementalWoocommerceStream):
+
+    def request_params(self, stream_state: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None, **kwargs):
+        params = super().request_params(stream_state=stream_state, next_page_token=next_page_token, **kwargs)
+
+        if not next_page_token and stream_state:
+            del params["after"]
+        return params
+
+    def read_records(
+            self, stream_state: Mapping[str, Any] = None, stream_slice: Optional[Mapping[str, Any]] = None, **kwargs
+        ) -> Iterable[Mapping[str, Any]]:
+
+            slice = super().read_records(sync_mode = SyncMode.full_refresh, stream_slice = stream_slice,stream_state = stream_state)
+            yield from self.filter_records_newer_than_state(stream_state=stream_state, records_slice=slice)
+
+
+class Coupons(IncrementalWoocommerceStream):
+    data_field = "coupons"
+    order_field = "date"
+
+    def path(self, **kwargs) -> str:
+        return f"{self.data_field}"
+
+class Customers(NonFilteredStream):
+    data_field = "customers"
+    order_field = "registered_date"
+
+    def path(self, **kwargs) -> str:
+        return f"{self.data_field}"
+
+
+class Orders(IncrementalWoocommerceStream):
+    data_field = "orders"
+    order_field = "date"
+
+    def path(self, **kwargs) -> str:
+        return f"{self.data_field}"
+
+# Source
+class SourceWoocommerce(AbstractSource):
+
+    def _convert_auth_to_token(self, username: str, password: str) -> str:
+        username = username.encode("latin1")
+        password = password.encode("latin1")
+        token = b64encode(b":".join((username, password))).strip().decode("ascii")
+        return token
+
+    def check_connection(self, logger, config) -> Tuple[bool, any]:
+        """
+        Testing connection availability for the connector.
+        """
+        shop = config["shop"]
+        api_version = "wc/v3"
+        api_key = config["api_key"]
```

Comment:
No need to define `api_key` & `api_password` & `headers` here, if you call them directly from `config` later.